ngraph/core/src/op/if.cpp

```diff
@@ -0,0 +1,323 @@
+    visitor.on_attribute("else_body", m_bodies[else_body_index]);
+    visitor.on_attribute("then_inputs", m_input_descriptions[then_body_index]);
+    visitor.on_attribute("else_inputs", m_input_descriptions[else_body_index]);
+    visitor.on_attribute("then_outputs", m_output_descriptions[then_body_index]);
+    visitor.on_attribute("else_outputs", m_output_descriptions[else_body_index]);
+    return true;
+}
+
+void op::v8::If::validate_and_infer_type_body(
+    std::shared_ptr<Function> body,
+    ngraph::op::util::MultiSubgraphInputDescriptionVector& input_descriptors)
+{
+    auto layer_id_map = std::map<size_t, std::shared_ptr<op::Parameter>>();
+    for (auto param : body->get_parameters())
+    {
+        layer_id_map.insert({param->get_instance_id(), param});
+    }
+    for (const auto& input_description : input_descriptors)
+    {
+        auto index = input_description->m_input_index;
+
+        auto body_parameter = body->get_parameters().at(input_description->m_body_parameter_index);
+        auto input_partial_shape = inputs().at(index).get_source_output().get_partial_shape();
+        if (input_partial_shape.is_static())
+        {
+            auto input_shape = input_partial_shape.to_shape();
+            Shape out_shape{input_shape};
+            body_parameter->set_partial_shape(out_shape);
+        }
+        else
+        {
+            body_parameter->set_partial_shape(input_partial_shape);
+        }
+    }
+    body->validate_nodes_and_infer_types();
+}
+void op::v8::If::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v8_If_validate_and_infer_types);
+    auto cond_output = inputs().at(0).get_source_output();
+
+    auto cond_partial_shape = cond_output.get_partial_shape();
+    auto cond_rank = cond_partial_shape.rank();
+    if (cond_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this, cond_rank.get_length() < 2, "Incorrect condition");
+    }
+    if (cond_partial_shape.is_static())
+    {
+        auto cond_shape = cond_partial_shape.to_shape();
+        if (cond_rank.get_length() == 1)
+        {
+            NODE_VALIDATION_CHECK(this, cond_shape.at(0) == 1, "Incorrect shape of condition");
+        }
+    }
+    // Trying to get cond as const value
+    if (const auto& cond_value = get_constant_from_source(cond_output))
+    {
+        auto val = cond_value->cast_vector<bool>();
+        auto cond_index = val[0] ? then_body_index : else_body_index;
+        auto body = m_bodies[cond_index];
+        auto input_descriptors = m_input_descriptions[cond_index];
+        validate_and_infer_type_body(body, input_descriptors);
+        auto output_nodes = outputs();
+        for (auto output_descr : m_output_descriptions[cond_index])
+        {
+            auto body_value =
+                body->get_results().at(output_descr->m_body_value_index)->input_value(0);
+            auto body_value_partial_shape = body_value.get_partial_shape();
+            set_output_type(output_descr->m_body_value_index,
```

Comment:
I guess `set_output_type(..., body_value.get_element_type(), body_value.get_partial_shape());` is enough
@jane-intel please confirm

Reply:
done